### PR TITLE
PLENTY-440_migrate_latest_3_9_2_changes

### PR DESCRIPTION
**Pull request by dblaichinger**
… use typescript components

### CODE REVIEW

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue [hunk 1]
```diff
@@ -38,17 +41,18 @@
         v-text="facet.noAvailableFiltersText"
       />
     </div>
-    <div v-else-if="facet.id === 'cat'">
+    <div v-else-if="facet.id === 'cat' && shouldShowCategoryFilter">
```
Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: The `faced.it === 'cat'` check is already included in `shouldShowCategoryFilter`

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue [hunk 2]
```diff
@@ -38,17 +41,18 @@
...
           <item-category-dropdown
             v-if="facet.findologicFilterType === 'select'"
+            :current-category="currentCategory"
             :facet="facet"
           />
         </div>
         <div
           v-for="value in facet.values"
-          v-else
+          v-else-if="facet.id === 'cat' && shouldShowCategoryFilter"
```
Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: The `facet.id === 'cat'` check is already included in `shouldShowCategoryFilter`

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue [hunk 3]
```diff
@@ -57,7 +61,7 @@
               :id="'option-' + value.id"
               class="form-check-input hidden-xs-up"
               type="checkbox"
-              :checked="value.selected"
+              :checked="isSelected"
```
Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: Where is `isSelected` defined? I can only find usages.

Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue:64]: Where is `isSelected` defined? I can only find usages.

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue [hunk 4]
```diff
@@ -86,7 +90,7 @@
                   :id="'option-' + subCategory.id"
                   class="form-check-input hidden-xs-up"
                   type="checkbox"
-                  :checked="subCategory.selected"
+                  :checked="isCategorySelected(subCategory)"
```
Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: Where is `isCategorySelected` defined?

Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue:93]: Where is `isCategorySelected` defined?

File: resources/js/src/app/components/itemList/filter/ItemCategoryDropdown.vue [hunk 5]
```diff
@@ -113,8 +116,61 @@ export default defineComponent({
...
+      }
+
+      return typeof UrlBuilder.getSelectedFilters().find(element => element.id === props.facet.id) !== 'undefined';
+    });
+
+    const isInCategoryPage = computed((): boolean => {
+      return typeof props.currentCategory !== 'undefined';
+    });
+
+    const getCategories = computed((): FacetValue[] | undefined  => {
```
Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/ItemCategoryDropdown.vue]: As this is a `computed`, I would not give it a getter name. Just `categories` would be enough.

Comment (TobiasGraml11) [inline resources/js/src/app/components/itemList/filter/ItemCategoryDropdown.vue:132]: As this is a `computed`, I would not give it a getter name. Just `categories` would be enough.

**Review by TobiasGraml11 — CHANGES_REQUESTED** [review]

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue (hunk 3 above)
Comment (dblaichinger) [reply resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: Good catch, not necessary anymore to handle cat filter in ItemFilter, we have ItemCategoryDropdown.

File: resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue (hunk 4 above)
Comment (dblaichinger) [reply resources/js/src/app/components/itemList/filter/FindologicItemFilter.vue]: Good catch, not necessary anymore to handle cat filter in ItemFilter, we have ItemCategoryDropdown.

**Review by TobiasGraml11 — APPROVED** [review]

**Pull request merged**